terror/terror.go

```diff
@@ -218,25 +217,23 @@ func (e *Error) getMsg() string {
 }
 
 // Gen generates a new *Error with the same class and code, and a new formatted message.
-func (e *Error) Gen(format string, args ...interface{}) *Error {
+func (e *Error) Gen(format string, args ...interface{}) error {
```

Comment:
how about change the function name to **"GenWithStack"** and change "GenByArgs" to **"GenWithStackByArgs"**?